### PR TITLE
Add Swift 3 swift-DEVELOPMENT-SNAPSHOT-2016-05-09 compatibility.

### PR DESCRIPTION
**Pull request by TheArtOfEngineering**
This version builds and passes tests on both Swift 3 DEVELOPMENT-SNAPSHOT-2016-05-09 and Swift 2.2.

Please be aware of the potential maintenance implications.

In Swift 3, by the first parameter is no longer removed by default and, unfortunately, you can’t #if #endif out just the function declaration, you have to do the entire function. From SE-0020, "Like other build configurations, #if swift isn't line-based - it encloses whole statements or declarations."

Given this, I erred on the side of maintaining your current public APIs, however since the philosophy of the first parameter has changed in Swift 3, you’ll probably want to look into updating the API.

As an example,
`class func encryptString(_ str: String, password: String) -> String? {`
would probably become
`class func encrypt(string: String, password: String) -> String? {`
in Swift 3.

So basically, the downside of the way that we can achieve compatibility is some functions are duplicated. Although, the syntax changes in Swift 3 are pretty far-reaching, so a number of the function bodies would have to change anyway.

Also please note, I've marked semantics changes with // Swift 3 Semantic Change.

### CODE REVIEW

**Comment by skreutzberger** [comment]
Tyler, thank you so much for your work and the explanation! Changing the way function declaration behaves is really a dramatic change and I feel a bit like moving from Python 2.7 to Python 3 😄  ... but we do not have a choice and should avoid the issue of Python where everyone is still stuck with the old version. So I will approve your PR and let's hope that Swift 3 quickly becomes the new standard.

About TravisCI: Do you know if we can test for Swift 2.2 and Swift 3?
